payment/src/main/java/org/killbill/billing/payment/core/PaymentProcessor.java

```diff
@@ -400,6 +400,12 @@ public boolean apply(final PaymentTransactionModelDao input) {
                         throw new PaymentApiException(ErrorCode.PAYMENT_INVALID_OPERATION, paymentStateContext.getTransactionType(), transactionToComplete.getTransactionStatus());
                     }
 
+                    // If Janitor confirmed the status, just return the status to the client.
+                    if (transactionType != TransactionType.AUTHORIZE &&
```

Comment:
because, for Authorizes, there will be some logic afterwards (in the running part) to handle the duplicate situation.